### PR TITLE
Add assertion structure

### PR DESCRIPTION
**Pull request by trevormcdonald**
## Proposed change/fix

- Add Assertion types and table
- Add `INGESTION` Org type
- Remove unnecessary clone in connection_pool
- Make new type definition for adding new organization
- Introduce if let syntax to destructure option
- remove `description` impl for error

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (could be a small readme update, documentation contribution, etc.)

## How to run/test

`clip`
`cargo build`
Test with the whole network using `docker-compose`

### CODE REVIEW

File: src/data_manager.rs [hunk 1]
```diff
@@ -48,8 +49,7 @@ impl DataManager {
         let conn = &*self.conn;
         conn.transaction::<_, _, _>(|| {
             let block_in_db = self.get_block_if_exists(block.block_num)?;
-            if block_in_db.is_some() {
-                let block_in_db = block_in_db.unwrap();
+            if let Some(block_in_db) = block_in_db {
```
Comment (Patrick-Erichsen) [inline src/data_manager.rs:52]: nice

File: src/models.rs [hunk 2]
```diff
@@ -284,3 +284,27 @@ pub struct NewAccreditation {
...
+    pub assertion_id: String,
+    pub assertor_pub_key: String,
+    pub assertion_type: AssertionTypeEnum,
+    pub object_id: String,
+    pub data_id: Option<String>,
+}
+
+#[derive(Debug, PartialEq, Queryable, Insertable)]
+#[table_name = "assertions"]
+pub struct NewAssertion {
```
Comment (Patrick-Erichsen) [inline src/models.rs:302]: could you explain why we use this `NewX` pattern / why its needed?

File: src/data_manager.rs [hunk 3]
```diff
@@ -376,6 +377,30 @@ impl DataManager {
...
+        for assertion in assertions {
+            self.update_assertion(&assertion.assertion_id, assertion.start_block_num)?;
+        }
+        diesel::insert_into(assertions::table)
+            .values(assertions)
+            .execute(&*self.conn)?;
+        Ok(())
+    }
+
+    fn update_assertion(
```
Comment (Patrick-Erichsen) [inline src/data_manager.rs:390]: I believe this is essentially a create or update? I know we use the `updateX` name elsewhere, but I feel like `create_or_update_assertion()` would be more descriptive.

Comment (trevormcdonald) [reply src/data_manager.rs:390]: Are you referring to `update_assertion`? That function only executes a `diesel::update` query.

**Pull request merged**